### DISCUSSION TITLE
Is Invoke-DbaDbShrink the same as "DBCC ShrinkFile('database_log', 0, TruncateOnly)"?

### DISCUSSION THREAD

**Discussion by DBAgnew** [dataplat/dbatools#7018] [Q&A]
Hello,

Is the command below the same as DBCC SHRINKFILE ('Database_log', 0 , TRUNCATEONLY)?
Invoke-DbaDbShrink -SqlInstance <ServerName> -FileType Log -ShrinkMethod TruncateOnly -Database <DatabaseName>

Thanks

**Comment by potatoqualitee** [comment]
The PowerShell code is `$file.Shrink(($desiredFileSize / 1024), $ShrinkMethod)` which is an SMO method. When I ran a XEvents Trace, this is what your command showed

`DBCC SHRINKFILE (N'anotherdb_log' , 4, TRUNCATEONLY)`

The `$desiredFileSize` is `$spaceUsed + $desiredSpaceAvailable`

**Reply by DBAgnew** [reply]
Thanks for feedback.
